carbon-design-system/carbon-for-ibm-dotcom
# fix(cloud-mega-menu): add style and polish the Cloud mega menu & keep platform visible on all bps

### Related Ticket(s)

DDS Ticket: N/A
HC JIRA Ticket: https://jsw.ibm.com/browse/HC-1813

### Description
- This PR implements styles and any other remaining polishing items for the Mega Menu design enhancements that are part of the "Unified Cloud Masthead + Mega Menu Project" (see project overview doc - https://ibm.ent.box.com/notes/775640792107).
- This is a follow-up PR to @annawen1's work for adding the baseline tabbing functionality to the cloud mega menu component - https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/5751 

### Design Spec
https://ibm.ent.box.com/file/774092317795

### Changelog

![Screen Shot 2021-04-21 at 9 58 17 AM](https://user-images.githubusercontent.com/1815714/115566285-1cf4f680-a288-11eb-90ec-d6d9f9dae4a7.png)

**New Components**

- `<dds-cloud-megamenu>` – an extended component of `<dds-megamenu>` that allows us to apply styles to the host component that's cloud design specific.
- `<dds-cloud-top-nav-name>` - – an extended component of `<dds-top-nav-name>` that allows us to apply styles to the host component that's cloud design specific, thus making this top nav name visible on all breakpoints.
-
**Removed**

- N/A

**Notes**
- There's a known issue within the mobile hamburger menu where the regular dropdown menu items are not showing up correctly. @annawen1 has opened a fix PR for this issue and the PR can be reviewed here - https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/5839

**Testing Instructions**
- Visit the `Cloud Masthead` Storybook variant - `/?path=/story/components-cloud-masthead--default`
- Ensure that the mega menu component matches with the design specs for all breakpoints.
